code/game/objects/items/storage/holsters.dm

```diff
@@ -300,6 +300,13 @@
 	if(istype(O,/obj/item/ammo_magazine/flamer_tank))
 		refuel(O, user)
 
+/obj/item/storage/holster/backholster/flamer/handle_item_insertion(obj/item/item, prevent_warning = 0, mob/user)
+	. = ..()
+	if (istype(item, /obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer))
+		var/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer/flamer = item
+		refuel(flamer.chamber_items[1], user)
+		flamer.update_ammo_count()
```

Comment:
you need to check if this has actually passed or not, otherwise it will do it even if it wasn't actually stored.
An easier way to do this is just call parent, then if holstered_item = w, do the thing.
it can only be the holstered item if it was holstered successfully.